src/commands/deployImage/imageSource/containerRegistry/acr/AcrRepositoriesListStep.ts

```diff
@@ -37,11 +38,18 @@ export class AcrRepositoriesListStep extends RegistryRepositoriesListStepBase {
             }
         }
 
+        const picks: QuickPickItem[] = [];
+        if (!repositoryNames.length) {
+            picks.push(noMatchingResourcesQp);
+        }
+
         // Preferring 'suppressPersistence: true' over 'priority: highest' to avoid the possibility of a double parenthesis appearing in the description
-        return repositoryNames.map((rn) => {
-            return !!suggestedRepository && rn === suggestedRepository ?
-                { label: rn, description: currentlyDeployed, suppressPersistence: true } :
-                { label: rn, suppressPersistence: srExists };
-        });
+        return picks.concat(
```

Comment:
Small thing, but you shouldn't need this concat here (since the picks array should be empty. I think you could probably just return an array with `[noMatchingResourcesQp]` on line 43 and leave the rest of the code unchanged.